lib/ansible/plugins/filter/network.py

```diff
@@ -395,6 +395,79 @@ def comp_type5(unencrypted_password, encrypted_password, return_orginal=False):
     return False
 
 
+def vlan_parser(vlan_list):
+
+    '''
+        Input: Unsorted list of vlan integers
+        Output: Sorted list of integers according to Cisco IOS vlan list rules
+
+        1. Vlans are listed in ascending order
+        2. Runs of 3 or more consecutive vlans are listed with a dash
+        3. The first line of the list can be 48 characters long
+        4. Subsequent list lines can be 44 characters
+    '''
+
+    # Sort and remove duplicates
+    sorted_list = sorted(set(vlan_list))
+
+    if sorted_list[0] < 1 or sorted_list[-1] > 4094:
+        raise AnsibleFilterError('Valid VLAN range is 1-4094')
+
+    parse_list = []
+    idx = 0
+    while idx < len(sorted_list):
+        start = idx
+        end = start
+        while end < len(sorted_list) - 1:
+            if sorted_list[end + 1] - sorted_list[end] == 1:
+                end += 1
+            else:
+                break
+
+        if start == end:
+            # Single VLAN
+            parse_list.append(str(sorted_list[idx]))
+        elif start + 1 == end:
+            # Run of 2 VLANs
+            parse_list.append(str(sorted_list[start]))
+            parse_list.append(str(sorted_list[end]))
+        else:
+            # Run of 3 or more VLANs
+            parse_list.append(str(sorted_list[start]) + '-' + str(sorted_list[end]))
+        idx = end + 1
+
+    line_count = 0
+    result = ['']
+    for vlans in parse_list:
+        # First line (" switchport trunk allowed vlan ")
+        if line_count == 0:
+            if len(result[line_count] + vlans) > 48:
```

Comment:
I think it would be great if this was configurable, and perhaps defaulted to this value? I suspect other similar OS's have slightly different numbers.

Reply:
@itdependsnetworks what aspect do you want configurable? This particular logic is built around IOS vlan lists specifically, I can't really comment if it applies to JunOS etc. Happy to rename to something besides 'vlan_parser' to indicate as such. 'ios_style_vlans' ?

Reply:
"what aspect do you want configurable": '{{set parsed_vlans = vlans | vlan_parser(first_line_len=48, other_line_count=40) }}`

Non-ios, but ios like devices such as eos, nxos, etc.. They all have similar structure. Additionally, it may work fine for your version, but another version may have 50 char's.

Reply:
I have incorporated the requested change.